server/norace_test.go

```diff
@@ -8601,12 +8601,6 @@ func TestNoRaceJetStreamClusterDifferentRTTInterestBasedStreamPreAck(t *testing.
 	mset, err := slow.GlobalAccount().lookupStream("EVENTS")
 	require_NoError(t, err)
 
-	// Make sure preAck is non-nil, so we know the logic has kicked in.
```

Comment:
Why did we remove this check?

Reply:
It has never done the right thing, even going back to the commit where the test was added a year ago. It's just that the bug in `require_NotNil` hid the fact it was `nil` at this point all along.

Reply:
So prechecks is always nil here? In the test?

Reply:
Yep. Notably nowhere else in the codebase looks at whether it's `nil`, they just do `len(...)` on it, which is what the test then goes on to do in `checkFor`.

Reply:
Then we need to slow down the server more since this is supposed to be testing that preacks work etc. We should not simply remove the check that it is not nil.

Reply:
Turns out this test was wrong in all sorts of ways:

1. The routes between S1 and S2 were only slow in one direction, not both;
2. The stream was listening on one subject but the publisher was publishing on a totally different subject, therefore the consumer never received any messages to ack;
3. The final `checkFor` loop didn't wait for all of the messages to have even arrived before judging whether preacks had happened or not.

I've now fixed it I think.